src/binder/textualidentityparser.cpp

```diff
@@ -417,12 +439,64 @@ namespace BINDER_SPACE
                 cursor++;
             }
 
-            if (dwFoundNumbers == static_cast<DWORD>(iRequiredVersionParts))
+            // Compat with .NET Framework:
+            // - If no value is provided, the version is considered invalid.
+            //   Example: "MyAssembly, Version="
+            // - If the major version is provided and the remaining components are valid, the version is considered valid.
+            //   Examples:
+            //     "MyAssembly, Version=1"
+            //     "MyAssembly, Version=1."
+            //     "MyAssembly, Version=1.."
+            //     "MyAssembly, Version=1..."
+            if (dwFoundNumbers == 0)
             {
-                pAssemblyVersion->SetFeatureVersion(dwNumbers[0], dwNumbers[1]);
-                pAssemblyVersion->SetServiceVersion(dwNumbers[2], dwNumbers[3]);
-                fIsValid = TRUE;
+                goto Exit;
             }
+
+            if (dwFoundNumbers < 2 || dwNumbers[0] == UnspecifiedNumber || dwNumbers[1] == UnspecifiedNumber)
+            {
+                // Compat with .NET Framework: If the major or minor version is unspecified, all components are considered
+                // unspecified. Set the major version to the unspecified value (this is a special value indicating that the
+                // whole version is unspecified, see AssemblyName::HaveAssemblyVersion).
+                //
+                // Examples:
+                //   "MyAssembly, Version=."
+                //   "MyAssembly, Version=.1"
+                //   "MyAssembly, Version=1"
+                //   "MyAssembly, Version=1."
+                dwNumbers[0] = UnspecifiedNumber;
+                dwNumbers[1] = 0;
+                dwNumbers[2] = 0;
+                dwNumbers[3] = 0;
+            }
+            else
+            {
+                // Difference from .NET Framework:
+                // - In .NET Framework, unspecified components retain the unspecified value. This special value is treated as
+                //   unspecified when loading an assembly, even if explicitly specified as 65535 as part of the version.
+                // - In .NET Core, any unspecified version component after the minor version is given a value of zero. When
+                //   loading an assembly, .NET Core will match and load an assembly that has a version that is >= the requested
```

Comment:
@kouvel - It turns out I forgot strongnaming was needed in the Assembly.Load examples.

I believe "1.1" means "1.1.any.any" while "1.1.0" means "1.1.0.any". In particular, if the assembly's version is "1.1.2.0", the former pattern successfuly loads it while the latter doesn't (refdef mismatch) (Just tested now on full framework.)